metadata/metadata.go

```diff
@@ -622,6 +727,46 @@ func (resource *featureVariantResource) Update(lookup ResourceLookup, updateRes
 	return nil
 }
 
+func (resource *featureVariantResource) IsEquivalent(other ResourceVariant) (bool, error) {
+	/**
+	Key Fields for a FeatureVariant are
+	- Name
+	- Source
+	- Function
+	- Provider
+	- Entity
+	- Owner
+	- Type
+	*/
+
+	otherVariant, ok := other.(*featureVariantResource)
+	if !ok {
+		return false, nil
```

Comment:
does it make sense to return an error that you are comparing two different types of resources?

Reply:
yeah I could do that, my initial thnking was, "well if they're different they're not equal, but that shouldn't even happen"

Reply:
actually no, because i'm allowing comparisons between all resource variants

Reply:
wait so a feature will be compared to a primary source?

Reply:
I'm saying any resource variant can be compared to another resource variant